lib/octokit/repository.rb

```diff
@@ -10,7 +10,10 @@ class Repository
     #
     # @return [Repository]
     def self.from_url(url)
-      Repository.new(URI.parse(url).path[1..-1])
+      new URI.parse(url).path[1..-1].
+        gsub('repos/','').
+        split('/')[0..1].
```

Comment:
You can save a little work by making this `split('/', 3)[0..1].`. That way there won't be any more splitting after the second `/`.